idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/MainKtQuickFixRegistrar.kt

```diff
@@ -27,6 +29,9 @@ class MainKtQuickFixRegistrar : KtQuickFixRegistrar() {
                 isRedundant = true
             )
         )
+
+        // ChangeVariableMutabilityFix
+        registerPsiQuickFix<PsiElement, KtFirDiagnostic.VarOverriddenByVal>(VAR_OVERRIDDEN_BY_VAL_FACTORY)
```

Comment:
Maybe not necessary at the moment, but I guess we eventually need, say, `ValOrVarNodeFix` so that, e.g., var-to-val suggestion can be created via the same factory.

Btw, I'm referring to the positioning strategy we use in the checkers: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/checkers/checkers-component-generator/src/org/jetbrains/kotlin/fir/checkers/generator/diagnostics/FirDiagnosticsList.kt#L240

Reply:
I'll look for opportunities to refactor the factory-creation methods. What is positioning strategy used for?

Reply:
It literally means where to report diagnostics. E.g., instead of picking up specific modifiers from a declaration, using, e.g., VISIBILITY_MODIFIER positioning strategy on a declaration, the diagnostic will be marked on the problematic visibility modifier of that declaration.